mp/src/game/shared/momentum/mom_rocket.cpp

```diff
@@ -0,0 +1,256 @@
+{
+    // Make invisible
+    SetModelName(NULL_STRING);
+    SetSolid(SOLID_NONE);
+    m_takedamage = DAMAGE_NO;
+
+    // Pull out a bit
+    if (pTrace->fraction != 1.0)
+    {
+        SetAbsOrigin(pTrace->endpos + (pTrace->plane.normal * 1.0f));
+    }
+
+    Vector vecOrigin = GetAbsOrigin();
+    CBaseEntity *pOwner = GetOwnerEntity();
+
+    float flDamage = GetDamage();
+    float flRadius = GetRadius();
+
+    // Create explosion effect with no damage
+    ExplosionCreate(vecOrigin, GetAbsAngles(), pOwner, flDamage, flRadius, false);
+
+    // Damage
+    CTakeDamageInfo info(this, pOwner, vec3_origin, vecOrigin, flDamage, GetDamageType());
+    RadiusDamage(info, vecOrigin, flRadius, CLASS_NONE, NULL);
+
+    if (m_hRocketTrail)
+    {
+        m_hRocketTrail->SetLifetime(0.1f);
+        m_hRocketTrail->SetParent(NULL);
+        m_hRocketTrail = NULL;
+    }
+
+    m_hOwner = NULL;
+
+    StopSound("Missile.Ignite");
+
+    // Remove the rocket
+    UTIL_Remove(this);
+}
+
+void CMomRocket::Touch(CBaseEntity *pOther)
+{
+    Assert(pOther);
+
+    // Don't touch triggers
+    if (pOther->IsSolidFlagSet(FSOLID_TRIGGER | FSOLID_VOLUME_CONTENTS))
+        return;
+
+    // Handle hitting skybox (disappear).
+    const trace_t *pTrace = &CBaseEntity::GetTouchTrace();
+    if (pTrace->surface.flags & SURF_SKY)
+    {
+        if (m_hRocketTrail)
+        {
+            m_hRocketTrail->SetLifetime(0.1f);
+            m_hRocketTrail->SetParent(NULL);
+            m_hRocketTrail = NULL;
+        }
+
+        m_hOwner = NULL;
+        StopSound("Missile.Ignite");
+        UTIL_Remove(this);
+        return;
+    }
+
+    // Explode
+    trace_t trace;
+    memcpy(&trace, pTrace, sizeof(trace_t));
+    Explode(&trace, pOther);
+}
+
+void CMomRocket::CreateSmokeTrail()
+{
+    if (m_hRocketTrail)
+        return;
+
+    // Smoke trail.
+    if ((m_hRocketTrail = RocketTrail::CreateRocketTrail()) != NULL)
+    {
+        m_hRocketTrail->m_Opacity = 0.2f;
+        m_hRocketTrail->m_SpawnRate = 100;
+        m_hRocketTrail->m_ParticleLifetime = 0.5f;
+        m_hRocketTrail->m_StartColor.Init(0.65f, 0.65f, 0.65f);
+        m_hRocketTrail->m_EndColor.Init(0.0, 0.0, 0.0);
+        m_hRocketTrail->m_StartSize = 8;
+        m_hRocketTrail->m_EndSize = 32;
+        m_hRocketTrail->m_SpawnRadius = 4;
+        m_hRocketTrail->m_MinSpeed = 2;
+        m_hRocketTrail->m_MaxSpeed = 16;
+
+        m_hRocketTrail->SetLifetime(999);
+        m_hRocketTrail->FollowEntity(this, "0");
+    }
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: Spawn a new rocket
+//
+// Input  : &vecOrigin -
+//          &vecAngles -
+//          *pentOwner -
+//
+// Output : CMomRocket
+//-----------------------------------------------------------------------------
+CMomRocket *CMomRocket::EmitRocket(const Vector &vecOrigin, const QAngle &vecAngles, CBaseEntity *pentOwner = NULL)
```

Comment:
```suggestion
CMomRocket *CMomRocket::EmitRocket(const Vector &vecOrigin, const QAngle &vecAngles, CBaseEntity *pentOwner /*= nullptr*/)
```

I like to block comment out function defaults like this, makes it easier to remember what they are, if there are any.